galley/pkg/config/analysis/diag/level.go

```diff
@@ -68,3 +69,36 @@ func GetUppercaseStringToLevelMap() map[string]Level {
 	}
 	return m
 }
+
+// MessageThreshold is a wrapper around Level to use as cobra command line arguments
+type MessageThreshold struct {
+	Level
+}
+
+// String satisfies interface pflag.Value
+func (m *MessageThreshold) String() string {
+	return m.Level.String()
+}
+
+// Type satisfies interface pflag.Value
+func (m *MessageThreshold) Type() string {
+	return "Level"
+}
+
+// Set satisfies interface pflag.Value
+func (m *MessageThreshold) Set(s string) error {
+	l, err := levelFromString(s)
+	if err != nil {
+		return err
+	}
+	m.Level = l
+	return nil
+}
+
+func levelFromString(s string) (Level, error) {
```

Comment:
"FromString" is generally "Parse" in golang, so "parseLevel" makes more sense to me.